src/dmd/backend/cod3.d

```diff
@@ -2158,11 +2157,11 @@ void cod3_ptrchk(ref CodeBuilder cdb,code *pcs,regm_t keepmsk)
        )
     {
         // Load the offset into a register, so we can push the address
-        idxregs = (I16 ? IDXREGS : ALLREGS) & ~keepmsk; // only these can be index regs
-        assert(idxregs);
-        allocreg(cdb,&idxregs,&reg,TYoffset);
+        regm_t idxregs2 = (I16 ? IDXREGS : ALLREGS) & ~keepmsk; // only these can be index regs
```

Comment:
In the previous version of the code `idxregs` was modified here, but used used below on line 2180:
https://github.com/dlang/dmd/blob/1fc01851310f3b1cf407ce010d57877d3cd4ac6e/src/dmd/backend/cod3.d#L2181

Was that modification of `idxregs` a bug, which this PR fixes by making a copy instead?

Reply:
It wasn't a bug. It is a distinct use of a variable, so it should be a separate variable.